src/pages/api/locale/index.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  Endpoint,
+  getServerSideCookies,
+  HttpStatus,
+  setAlertMessagesCookie,
+} from "@fwl/web";
+import {
+  loggingMiddleware,
+  wrapMiddlewares,
+  tracingMiddleware,
+  errorMiddleware,
+  recoveryMiddleware,
+  rateLimitingMiddleware,
+  Middleware,
+} from "@fwl/web/dist/middlewares";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { DynamoUser } from "@src/@types/dynamo-user";
+import { Handler } from "@src/@types/handler";
+import { UserCookie } from "@src/@types/user-cookie";
+import { getAndPutUser } from "@src/commands/get-and-put-user";
+import { configVariables } from "@src/configs/config-variables";
+import { logger } from "@src/configs/logger";
+import rateLimitingConfig from "@src/configs/rate-limiting-config";
+import getTracer from "@src/configs/tracer";
+import { NoUserCookieFoundError } from "@src/errors/errors";
+import { ERRORS_DATA, webErrorFactory } from "@src/errors/web-errors";
+import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
+import { getDBUserFromSub } from "@src/queries/get-db-user-from-sub";
+import { generateAlertMessage } from "@src/utils/generate-alert-message";
+import { AVAILABLE_LANGUAGE } from "@src/utils/get-locale";
+
+const getHandler: Handler = (request, response): Promise<void> => {
+  const webErrors = {
+    databaseUnreachable: ERRORS_DATA.DATABASE_UNREACHABLE,
+  };
+
+  return getTracer().span("get-locale handler", async (span) => {
+    const userCookie = await getServerSideCookies<UserCookie>(request, {
+      cookieName: "user-cookie",
+      isCookieSealed: true,
+      cookieSalt: configVariables.cookieSalt,
+    });
+
+    if (!userCookie) {
+      span.setDisclosedAttribute("user cookie found", false);
+      throw new NoUserCookieFoundError();
+    }
+    span.setDisclosedAttribute("user cookie found", true);
+
+    const currentDBUser = await getDBUserFromSub(userCookie.sub).catch(
+      (error) => {
+        span.setDisclosedAttribute("user locale retrieved", false);
+        throw webErrorFactory({
+          ...webErrors.databaseUnreachable,
+          parentError: error,
+        });
+      },
+    );
+
+    if (!currentDBUser) {
+      span.setDisclosedAttribute("user locale value", null);
+      response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+      response.setHeader("location", "/");
+      response.end();
+      return;
+    }
+
+    span.setDisclosedAttribute("user locale retrieved", true);
+    span.setDisclosedAttribute("user locale value", currentDBUser.locale);
+    response.statusCode = HttpStatus.OK;
+    response.json({ locale: currentDBUser.locale });
```

Comment:
Following REST convention we should only send back the resource and not embed the data in an object with a `locale` key.
```suggestion
    response.json(currentDBUser.locale);
```

Reply:
I did this because using `response.json(currentDBUser.locale);` throws a `JSON SyntaxError` because of the nested accessed value inside the object. I would need to declare a variable beforehand to do so. But I can do it if you want to.

Reply:
What do you mean?
Isn't it just a string like "fr" or "en"?

Reply:
```ts
// Throws
response.json(currentDBUser.locale);

// Works
const updatedLocale = currentDBUser.locale;
response.json(updatedLocale);
```

Reply:
😮 What is this wizardry? 🤔
And in both cases, is `currentDBUser.locale` has a value?
If the locale should be "en" by default, maybe this could do the trick?
```ts
response.json(currentDBUser.locale || "en");
```
(sorry, but i'm very curious on that partlicuar magic 😁)

Reply:
>  also get `SyntaxError: Unexpected token e in JSON at position 0.`. Only returning an object is working 🤯 .

Maybe with `response.end` instead, like we do in the Profile's ping handler?

```ts
export function pingHandler(tracer: Tracer) {
  return (
    request: IncomingMessage,
    response: ServerResponse,
  ): Promise<void> => {
    return tracer.span("ping-handler", async () => {
      response.statusCode = HttpStatus.OK;
      response.end("OK"); 👈👈👈
    });
  };
}
```

Reply:
Not working :/. Would you be ok to fetch the branch and verify I'm not crazy? Haha.
I just pulled the switch from POST to PATCH 👍 .

Reply:
I'm ok, I'm on it